api/task_processor/models.py

```diff
@@ -90,44 +101,37 @@ class Meta:
     def create(
         cls,
         task_identifier: str,
+        scheduled_for: datetime,
+        priority: TaskPriority = TaskPriority.NORMAL,
+        queue_size: int = None,
         *,
         args: typing.Tuple[typing.Any] = None,
         kwargs: typing.Dict[str, typing.Any] = None,
     ) -> "Task":
+        if queue_size:
+            if cls.is_queue_full(task_identifier, queue_size):
+                raise TaskQueueFullError(
+                    f"Queue for task {task_identifier} is full. "
+                    f"Max queue size is {queue_size}"
+                )
         return Task(
             task_identifier=task_identifier,
+            scheduled_for=scheduled_for,
+            priority=priority,
             serialized_args=cls.serialize_data(args or tuple()),
             serialized_kwargs=cls.serialize_data(kwargs or dict()),
         )
 
     @classmethod
-    def schedule_task(
-        cls,
-        schedule_for: datetime,
-        task_identifier: str,
-        queue_size: typing.Optional[int],
-        *,
-        args: typing.Tuple[typing.Any] = None,
-        kwargs: typing.Dict[str, typing.Any] = None,
-    ) -> "Task":
-        if queue_size:
-            if (
-                cls.objects.filter(
-                    task_identifier=task_identifier, completed=False, num_failures__lt=3
-                ).count()
-                > queue_size
-            ):
-                raise TaskQueueFullError(
-                    f"Queue for task {task_identifier} is full. "
-                    f"Max queue size is {queue_size}"
-                )
-        task = cls.create(
-            task_identifier=task_identifier,
-            args=args,
-            kwargs=kwargs,
+    def is_queue_full(cls, task_identifier: str, queue_size: int) -> bool:
```

Comment:
This can probably be marked as private now, right? 

```suggestion
    def _is_queue_full(cls, task_identifier: str, queue_size: int) -> bool:
```